lib/lndclient/LndClient.ts

```diff
@@ -38,24 +38,30 @@ class LndClient extends BaseClient {
    */
   constructor(config: LndClientConfig) {
     super(Logger.lnd);
+    let shouldConnect: boolean = true;
     const { disable, certpath, macaroonpath } = config;
 
     if (disable) {
-      this.setStatus(ClientStatus.DISABLED);
-    } else if (!fs.existsSync(certpath)) {
+      shouldConnect = false;
+    }
+    if (!fs.existsSync(certpath)) {
       this.logger.error('could not find lnd certificate, is lnd installed?');
-      this.setStatus(ClientStatus.DISABLED);
-    } else if (!fs.existsSync(macaroonpath)) {
+      shouldConnect = false;
+    }
+    if (!fs.existsSync(macaroonpath)) {
       this.logger.error('could not find lnd macaroon, is lnd installed?');
-      this.setStatus(ClientStatus.DISABLED);
-    } else {
+      shouldConnect = false;
+    }
+    if (shouldConnect) {
```

Comment:
What's the appeal of using this pattern of `shouldConnect` instead of a chain of if-elses as before? I suppose this way we can log errors for both `macaroonpath` and `certpath` in one go which is nice, but just want to make sure I understand the changes here. If we do keep this pattern, I think the flag should be renamed to `shouldEnable`.

Reply:
code is more readable.
multiple checks in one run

renamed.

done